ci/scripts/driver.sh

```diff
@@ -57,17 +57,26 @@ set -x
 ############################################################
 # query repo and get list of open PRs with tags {machine}-CI
 ############################################################
-pr_list_file="open_pr_list"
-touch "${GFS_CI_ROOT}/${pr_list_file}"
-list=$(${GH} pr list --repo "${REPO_URL}" --label "CI-${MACHINE_ID^}-Ready" --state "open")
-list=$(echo "${list}" | awk '{print $1;}' >> "${GFS_CI_ROOT}/${pr_list_file}")
-
-if [[ -s "${GFS_CI_ROOT}/${pr_list_file}" ]]; then
- pr_list=$(cat "${GFS_CI_ROOT}/${pr_list_file}")
-else
- echo "no PRs to process .. exit"
- exit 0
-fi 
+pr_list_dbfile="${GFS_CI_ROOT}/open_pr_list.db"
+if [[ ! -f "${pr_list_dbfile}" ]]; then
+  "${HOMEgfs}/ci/scripts/pr_list_database.py" --create "${pr_list_dbfile}"
+fi
+
+pr_list=$(${GH} pr list --repo "${REPO_URL}" --label "CI-${MACHINE_ID^}-Ready" --state "open" | awk '{print $1}') || true
```

Comment:
Yes, your suspicion is correct in that we actually do **_not_** want to exit here.  L65 is getting the appropriate PR list from **GitHub** based on labeling.  But once the list is obtained from **GitHub** it populates the data base and resets the label later once it advances and then the database handles the state.  This separation allows it to operate better as `cron` job and I control the system's state directly.  Notice that it will exit later (with a zero) when the data base is void if `driver.sh` really has nothing to do.